Add dropdown to hud header

clee2000 opened the pull request:
Referenced css from https://blog.logrocket.com/how-create-multilevel-dropdown-menu-react/ and its source code at https://github.com/Ibaslogic/react-multilevel-dropdown-menu

On my smaller screen, the HUD header was becoming a bit full, so I added a dropdown and moved a lot of the dev infra links into it.  I left metrics in the bar since I think it is used frequently, but I am open to moving links around if people want fewer clicks for pages they visit often.  

The dropdown component is reusable so if we also want to add a lot of repositories and the header doesn't have space for that, we can move them all into a dropdown as well.  I have no idea how the dropdown reacts to having so many things in it it takes up the entire page but that's a problem for future me (probably either make it tiered or scrollable).

Also remove the main branch link since the homepage is main.

Please visit the vercel preview to see if you like it.

Fixes https://github.com/pytorch/test-infra/issues/4439

huydhn reviewed (approved):
Nice!  I think "Nightly" could go under `Dev Infra` dropdown list too as mainly us look into that (cc @atalman)

ZainRizvi reviewed (approved):
Please consider nixing or hiding the "Help" and "Requests" buttons in favor of KPIs

clee2000 commented:
> Nice! I think "Nightly" could go under Dev Infra dropdown list too as mainly us look into that (cc @atalman)

Moved, also changed the name to Nightly Branch since there's also a Nightly Dashboard thats under the dropdown.

As a side note, did you know you can type in the blue boxes displaying the repo name and branch?  I remember learning that a while ago and I find it super helpful, but I feel like its really obscure. 

> Please consider nixing or hiding the "Help" and "Requests" buttons in favor of KPIs

Moved KPIs back to the header, but I didn't get rid of Help or Requests because I feel like it would be a bad idea to get rid of Help, and while Requests is barely used, it could just be because people changed the title or people couldn't find it because there were too many things in the header.

If anyone wants to continue moving things around, feel free to make more PRs, it should be really easy to change the dropdown
